### ISSUE TITLE
Enable nullable reference types for all projects

### ISSUE THREAD

**Issue by KvanTTT** [naudio/NAudio#825]
https://docs.microsoft.com/en-us/dotnet/csharp/nullable-references

It may help to decrease number of NRE

[NullabilityInference](https://github.com/icsharpcode/NullabilityInference) can be used at the beginning for decreasing manual actions.

**Comment by markheath** [comment]
I agree with this suggestion, and I've done a lot of code modernization for NAudio 3, but haven't got round to this yet (although the Vst3 project has it set). Will probably look to incrementally add this in but likely after NAudio 3 ships
